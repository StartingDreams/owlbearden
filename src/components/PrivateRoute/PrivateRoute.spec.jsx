import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { PrivateRoute, mapStateToProps } from './PrivateRoute';

jest.mock('react-router-dom', () => ({
  Redirect: 'Redirect',
  Route: props => (props.render(props)),
  withRouter: jest.fn(),
}));

describe('<PrivateRoute />', () => {
  it('mapStateToProps - isAuthenticated should be passed in', () => {
    const state = {
      auth: fromJS({ isAuthenticated: true }),
    };
    const props = mapStateToProps(state);
    expect(props.isAuthenticated).toBe(true);
  });
  it('authenticated users should match the dashboard snapshot', () => {
    const tree = renderer.create(<PrivateRoute
      isAuthenticated
      component={() => (<div>Authenticated</div>)}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('unauthenticated users should match the login snapshot', () => {
    const tree = renderer.create(<PrivateRoute
      isAuthenticated={false}
      component={() => (<div>Not Authenticated</div>)}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

