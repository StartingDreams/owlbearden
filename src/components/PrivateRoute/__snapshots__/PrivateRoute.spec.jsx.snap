// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> authenticated users should match the dashboard snapshot 1`] = `
<div>
  Authenticated
</div>
`;

exports[`<PrivateRoute /> unauthenticated users should match the login snapshot 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/login",
    }
  }
/>
`;
